test/runner/Unit/TestTensor.idr

```diff
@@ -218,76 +215,57 @@ iotaExamples = fixedProperty $ do
 partial
 show : Device => Property
 show = fixedProperty $ do
-  let x : Graph $ Tensor [] S32 = 1
-  show x === """
-    MkFn {parameters = [], root = 0, env =
-        0    FromLiteral [] 4
-      }
-    """
+  let x : Graph $ Tensor [] S32 = pure 1
+  show x === "[] => Lit [] 4"
 
-  let x : Graph $ Tensor [] S32 = 1 + 2
-  show x === """
-    MkFn {parameters = [], root = 2, env =
-        0    FromLiteral [] 4
-        1    FromLiteral [] 4
-        2    Add 0 1
-      }
-    """
+  let x : Graph $ Tensor [] S32 = pure $ 1 + 2
+  show x === "[] => Add (Lit [] 4) (Lit [] 4)"
 
-  let x = tensor {dtype = F64} [1.3, 2.0, -0.4]
-  show x === """
-    MkFn {parameters = [], root = 0, env =
-        0    FromLiteral [3] 12
+  let x : Graph _ = pure $ tensor {dtype = F64} [1.3, 2.0, -0.4]
+  show x === "[] => Lit [3] 12"
+
+  let x : Graph (Tensor [] S32) = do
+        y <- share $ 1 + 2
+        pure (y + y)
+  show x ===
+    """
+    [] => Add (Var 0) (Var 0), with vars {
+        0    Add (Lit [] 4) (Lit [] 4)
       }
     """
 
-  let x : Graph _ = do
-    y <- reduce @{Sum} [0] !(tensor {dtype = F64} [1.0, 2.0])
-    z <- cond !(tensor True) (map (\x => pure x + 11.0)) !21.0 pure !0.01
-    pure y + pure z
+  let f : Tensor [] S32 -> Graph $ Tensor [] S32
+      f x = do
+        z <- share $ the (Tensor [] S32) 5
+        map (\v => do v <- share v; pure $ v + v + z) (x + z)
+
+      x : Graph (Tensor [] S32) := do
+        x <- share =<< reduce @{Sum} [0] (tensor {dtype = S32} [1, 2])
+        pure $ x + !(map f x)
   show x === """
-    MkFn {parameters = [], root = 17, env =
-        0     FromLiteral [2] 12
-        4     FromLiteral [] 12
-        5     Broadcast {from = [], to = []} 4
-        6     Reduce {op = MkFn {parameters = [(1, MkParameter [] _), (2, MkParameter [] _)], root = 3, env =
-            1    Arg 1
-            2    Arg 2
-            3    Add 1 2
-          }, identity = 5, axes = [0]} 0
-        7     FromLiteral [] 1
-        8     FromLiteral [] 12
-        9     FromLiteral [] 12
-        16    Cond {predicate = 7, onTrueFn = MkFn {parameters = [(10, MkParameter [] _)], root = 14, env =
-            10    Arg 10
-            14    Map {f = MkFn {parameters = [(11, MkParameter [] _)], root = 13, env =
-                11    Arg 11
-                12    FromLiteral [] 12
-                13    Add 11 12
-              }} [10]
-          }, onTrueArg = 8, onFalseFn = MkFn {parameters = [(15, MkParameter [] _)], root = 15, env =
-            15    Arg 15
-          }, onFalseArg = 9}
-        17    Add 6 16
+    [] => Add (Var 2) (Map {f = [(3, [] 4)] => Map {f = [(5, [] 4)] => Add (Add (Arg 5) (Arg 5)) (Var 4)} [Add (Arg 3) (Var 4)], with vars {
+          4    Lit [] 4
+        }} [Var 2]), with vars {
+        2    Reduce {op = [(0, [] 4), (1, [] 4)] => Add (Arg 0) (Arg 1), identity = Broadcast {from = [], to = []} (Lit [] 4), axes = [0]} (Lit [2] 4)
       }
```

Comment:
this appears to be correct. I can only imagine this is a bug or limitation in XLA, or a misunderstanding of how to use their API (I _really really_ hope not)